mirascope/core/gemini/stream.py

```diff
@@ -35,7 +44,42 @@ def cost(self) -> float | None:
     def _construct_message_param(
         self, tool_calls: list[FunctionCall] | None = None, content: str | None = None
     ) -> ContentDict:
-        return {
-            "role": "model",
-            "parts": [{"type": "text", "text": content}] + (tool_calls or []),
-        }
+        """Constructs the message parameter for the assistant.
+        Gemini currently does not support tool calls for streaming.
+        """
+        return {"role": "model", "parts": [{"text": content}]}  # type: ignore
+
+    def construct_call_response(self) -> GeminiCallResponse:
+        if self.message_param is None:
+            raise ValueError(  # pragma: no cover
+                "No stream response, check if the stream has been consumed."
+            )
+        response = GenerateContentResponseType.from_response(
+            GenerateContentResponse(
+                candidates=[
+                    Candidate(
+                        finish_reason=self.finish_reasons[0]
+                        if self.finish_reasons
+                        else 1,
+                        content=Content(
+                            role=self.message_param["role"],
+                            parts=self.message_param["parts"],
+                        ),
+                    )
+                ]
+            )
+        )
+        return GeminiCallResponse(
+            metadata=self.metadata,
+            response=response,
+            tool_types=self.tool_types,
+            prompt_template=self.prompt_template,
+            fn_args=self.fn_args if self.fn_args else {},
+            dynamic_config=self.dynamic_config,
+            messages=self.messages,  # type: ignore
```

Comment:
again, don't love the type ignore if we can find a way around it, and `self.messages` should be the right type, and if it's not we should fix that

Reply:
done